### PR TITLE
Fixes an issue when generating a type erasure exception in TypeExtractor.

### PR DESCRIPTION
**Pull request by twalthr**
The bug has been mentioned in #842.

A additional test case tests the fix.

### CODE REVIEW

File: stratosphere-java/src/main/java/eu/stratosphere/api/java/typeutils/TypeExtractor.java [hunk 1]
```diff
@@ -413,9 +413,11 @@ private static Type getParameterType(Class<?> baseClass, ArrayList<Type> typeHie
 					
 					// variable could not be determined
 					if (tupleSubTypes[i] == null) {
-						throw new InvalidTypesException("Type of TypeVariable '" + ((TypeVariable<?>) t).getName() + "' in '"
-								+ ((TypeVariable<?>) t).getGenericDeclaration()
-								+ "' could not be determined. This is most likely a type erasure problem.");
+						throw new InvalidTypesException("Type of TypeVariable '" + ((TypeVariable<?>) subtypes[i]).getName() + "' in '"
+								+ ((TypeVariable<?>) subtypes[i]).getGenericDeclaration()
+								+ "' could not be determined. This is most likely a type erasure problem."
```
Comment (uce) [inline stratosphere-java/src/main/java/eu/stratosphere/api/java/typeutils/TypeExtractor.java:418]: Missing whitespace after `.`

File: stratosphere-java/src/main/java/eu/stratosphere/api/java/typeutils/TypeExtractor.java [hunk 2]
```diff
@@ -449,7 +451,7 @@ else if (t instanceof TypeVariable) {
 					return typeInfo;
 				} else {
 					throw new InvalidTypesException("Type of TypeVariable '" + ((TypeVariable<?>) t).getName() + "' in '"
-							+ ((TypeVariable<?>) t).getGenericDeclaration() + "' could not be determined. "
+							+ ((TypeVariable<?>) t).getGenericDeclaration() + "' could not be determined. This is most likely a type erasure problem."
```
Comment (uce) [inline stratosphere-java/src/main/java/eu/stratosphere/api/java/typeutils/TypeExtractor.java:454]: Missing whitespace after `.`

File: stratosphere-java/src/test/java/eu/stratosphere/api/java/type/extractor/TypeExtractorTest.java [hunk 3]
```diff
@@ -1304,4 +1304,27 @@ public String map(Writable value) throws Exception {
...
+		@Override
+		public void flatMap(Tuple2<A, B> value, Collector<Tuple2<C, D>> out) throws Exception {
+			
+		}
+		
+	}
+	
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	@Test
+	public void testTypeErasureException() {
```
Comment (uce) [inline stratosphere-java/src/test/java/eu/stratosphere/api/java/type/extractor/TypeExtractorTest.java:1320]: We can simplify this to

``` java
@SuppressWarnings({"unchecked", "rawtypes"})
@Test(expected = InvalidTypesException.class)
public void testTypeErasureException() {
    TypeExtractor.getFlatMapReturnTypes(...);
}
```

**Comment by rmetzger** [comment]
I don't understand why you want whitespaces after the dot?

**Comment by uce** [comment]
The exception message is:

```
This is most likely a type erasure problem.The type extraction ...
```

A single whitespace after period (Satzende) is pretty standard, no? If you just look at the diff comment, you don't see how the message continues. Is that's why you are asking?

This is a minor style comment, so let's not discuss this to death and feel free to keep it that way.

**Comment by twalthr** [comment]
@uce I agree, I add the whitespace later.

**Comment by StephanEwen** [comment]
Merging and will fix the whitespace issue

**Comment by StephanEwen** [comment]
Merged in 01a17b7071590563f124e822856d7bb66c265a8b

**Pull request closed without merging**